### PR TITLE
Allow to create talk rooms from calendar

### PR DESCRIPTION
**Pull request by georgehrke**
Signed-off-by: Georg Ehrke <developer@georgehrke.com>

### CODE REVIEW

**Comment by tcitworld** [comment]
We should add the URI to the [`CONFERENCE`](https://tools.ietf.org/html/rfc7986#section-5.11) property as well.

**Comment by georgehrke** [comment]
I tried that, there is a problem with the encoding of FEATURE param in ical.js

Have to fix / report that upstream. :)

File: src/components/Editor/Invitees/InviteesList.vue [hunk 1]
```diff
@@ -161,6 +191,30 @@ export default {
...
 			})
 		},
+		async createTalkRoom() {
+			try {
+				this.creatingTalkRoom = true
+				const url = await createTalkRoom(this.calendarObjectInstance.title)
+
+				let newDescription
+				if (!this.calendarObjectInstance.description) {
+					newDescription = url
```
Comment (tcitworld) [inline src/components/Editor/Invitees/InviteesList.vue]: Add a new line at the end so it's easier to edit afterwards.

File: src/components/Editor/Invitees/InviteesList.vue [hunk 2]
```diff
@@ -161,6 +191,30 @@ export default {
...
+				} else {
+					newDescription = this.calendarObjectInstance.description + '\r\n\r\n' + url
```
Comment (tcitworld) [inline src/components/Editor/Invitees/InviteesList.vue]: Probably here as well.

File: lib/Controller/ViewController.php [hunk 3]
```diff
@@ -76,6 +86,7 @@ public function index():TemplateResponse {
 			'show_weekends' => $this->config->getUserValue($this->userId, $this->appName, 'showWeekends', 'yes') === 'yes',
 			'show_week_numbers' => $this->config->getUserValue($this->userId, $this->appName, 'showWeekNr', 'no') === 'yes',
 			'skip_popover' => $this->config->getUserValue($this->userId, $this->appName, 'skipPopover', 'no') === 'yes',
+			'talk_enabled' => $this->appManager->isEnabledForUser('spreed'),
```
Comment (rullzer) [inline lib/Controller/ViewController.php]: It really is time to itnroduce you to the initialstate api ;) @ChristophWurst we must spread the word of it!

Comment (georgehrke) [reply lib/Controller/ViewController.php]: Yes, it's on my never ending todo list ;)

File: src/components/Editor/Invitees/InviteesList.vue (hunk 1 above)
Comment (georgehrke) [reply src/components/Editor/Invitees/InviteesList.vue]: Resolved

File: src/components/Editor/Invitees/InviteesList.vue (hunk 2 above)
Comment (georgehrke) [reply src/components/Editor/Invitees/InviteesList.vue]: Resolved

**Comment by georgehrke** [comment]
`CONFERENCE` property and initial state thing is something i would like to address later :)

File: src/services/talkService.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+import HTTPClient from '@nextcloud/axios'
+import { translate as t } from '@nextcloud/l10n'
+import { generateUrl, generateOcsUrl } from '@nextcloud/router'
+
+export function isTalkEnabled() {
```
Comment (tcitworld) [inline src/services/talkService.js]: Remove me

**Pull request merged**